.github/workflows/test.yml

```diff
@@ -1,27 +1,54 @@
 name: Testing
 
-on: [push, pull_request]
+on: [pull_request]
 jobs:
   test:
     timeout-minutes: 360
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [macOS-latest, ubuntu-latest, windows-latest]
-        ghc: ['8.10.2', '8.8.4', '8.6.5']
+        # all versions to only build or test for non windows os's
+        # inclusions will modify the major ones to mark them as testables
+        os: [macOS-latest, ubuntu-latest]
+        ghc: ['8.10.2', '8.10.1', '8.8.4', '8.8.3', '8.8.2', '8.6.5', '8.6.4']
         ghc-lib: [false]
-        exclude:
-          - os: windows-latest
-            ghc: '8.10.2' # broken due to https://gitlab.haskell.org/ghc/ghc/-/issues/18550
-          - os: windows-latest
-            ghc: '8.8.4' # also fails due to segfault :(
         include:
```

Comment:
this is the less worse version i got to work, feel free to suggest another better one :slightly_smiling_face: 
//cc @pepeiborra @ndmitchell